socorro/unittest/cron/jobs/test_archivescraper.py

```diff
@@ -0,0 +1,313 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import os
+
+from jinja2 import Environment
+import psycopg2
+import pytest
+import requests_mock
+
+from socorro.cron.crontabber_app import CronTabberApp
+from socorro.cron.jobs.archivescraper import ArchiveScraperCronApp
+from socorro.unittest.cron.crontabber_tests_base import get_config_manager
+
+
+HOST = 'http://archive.example.com'
+
+INDEX_PAGE_TEMPLATE = """
+<!DOCTYPE html>
+<html>
+    <head>
+        <meta charset="UTF-8">
+        <title>Directory Listing: {{ path }}</title>
+    </head>
+    <body>
+        <h1>Index of {{ path }}</h1>
+        <table>
+            <tr>
+                <th>Type</th>
+                <th>Name</th>
+                <th>Size</th>
+                <th>Last Modified</th>
+            </tr>
+            {% for link in links %}
+            {% set is_dir = link.name.endswith('/') %}
+            <tr>
+                <td>{% if is_dir %}Dir{% else %}File{% endif %}</td>
+                <td><a href="{{ link.path }}">{{ link.name }}</a></td>
+                <td>{% if not is_dir %}18K{% endif %}</td>
+                <td>{% if not is_dir %}26-Oct-2018 03:14{% endif %}</td>
+            </tr>
+            {% endfor %}
+        </table>
+    </body>
+</html>
+"""
+
+
+def absolute_links_parents(path):
+    """Generate (path, link name) for all parents of a given path
+
+    :arg path: a url path
+
+    :returns: iterator over all the (parent path, child link) pairs
+
+    For example:
+
+    >>> list(absolute_links_parents('/pub/firefox/candidates/64.0b13-candidates/'))
+    [('/', 'pub/'), ('/pub/', 'firefox/'), ('/pub/firefox/', 'candidates/'),
+    ('/pub/firefox/candidates/', '64.0b13-candidates/')]
+
+    """
+    path_parts = path.strip('/').split('/')
+    for i, path_part in enumerate(path_parts):
+        parent_path = '/' + '/'.join(path_parts[0:i])
+        if parent_path != '/':
+            parent_path += '/'
```

Comment:
This is interesting. archive.mozilla.org doesn't redirect from `dir` to `dir/`, so if you go to `dir`, you get a 404. So for dir pages, we only want `requests_mock` to cover the path with a `/` at the end.